src/setup_payload/tests/TestQRCode.cpp

```diff
@@ -117,54 +118,77 @@ void TestPayloadBase38Rep(nlTestSuite * inSuite, void * inContext)
 
 void TestBase38(nlTestSuite * inSuite, void * inContext)
 {
-    uint8_t input[] = { 10, 10, 10 };
+    uint8_t input[3] = { 10, 10, 10 };
+    char encodedBuf[32];
+    MutableByteSpan inputSpan(input);
+    MutableCharSpan encodedSpan(encodedBuf);
 
     // basic stuff
-    NL_TEST_ASSERT(inSuite, base38Encode(input, 0).empty());
-    NL_TEST_ASSERT(inSuite, base38Encode(input, 1) == "A0");
-    NL_TEST_ASSERT(inSuite, base38Encode(input, 2) == "OT10");
-    NL_TEST_ASSERT(inSuite, base38Encode(input, 3) == "-N.B0");
+    base38Encode(inputSpan.SubSpan(0, 0), encodedSpan);
+    NL_TEST_ASSERT(inSuite, strlen(encodedBuf) == 0);
+    base38Encode(inputSpan.SubSpan(0, 1), encodedSpan);
+    NL_TEST_ASSERT(inSuite, strcmp(encodedBuf, "A0") == 0);
+    base38Encode(inputSpan.SubSpan(0, 2), encodedSpan);
+    NL_TEST_ASSERT(inSuite, strcmp(encodedBuf, "OT10") == 0);
+    base38Encode(inputSpan, encodedSpan);
+    NL_TEST_ASSERT(inSuite, strcmp(encodedBuf, "-N.B0") == 0);
+
+    // test null termination of output buffer
```

Comment:
Add a test to validate explicit presence of null termination by doing:

```
// Force no nulls in output buffer
memset(encodedSpan.data(), '?', encodedSpan.size()); 

size_t encoded_len = strnlen(encodedSpan.data(), sizeof(encoded_buf));
NL_TEST_ASSERT(inSuite, encoded_len == strlen("A0"));
```

If null termination does not occur, strnlen will yield max size.